lib/mayaUsd/fileio/utils/adaptor.cpp

```diff
@@ -58,8 +88,26 @@ UsdMayaAdaptor::UsdMayaAdaptor(const MObject& obj)
 {
 }
 
+UsdMayaAdaptor::UsdMayaAdaptor(const MObject& obj, const UsdMayaJobExportArgs* jobExportArgs)
+    : _handle(obj)
+    , _jobExportArgs(jobExportArgs)
+{
+}
+
+UsdMayaAdaptor::UsdMayaAdaptor(
+    const UsdMayaPrimReaderArgs& jobImportArgs,
+    UsdMayaPrimReaderContext&    context)
+    : _jobImportArgs(&jobImportArgs)
+    , _jobImportContext(&context)
+{
+}
+
 UsdMayaAdaptor::operator bool() const
 {
+    if (_jobImportArgs && _jobImportContext) {
```

Comment:
Can you document logic in here? I will take a few cycles to understand all the modes adaptor can be in.